packages/wallet/frontend/src/pages/auth/reset/[token].tsx

```diff
@@ -0,0 +1,157 @@
+                  openDialog(
+                    <SuccessDialog
+                      onClose={closeDialog}
+                      title="New password set."
+                      content="Please login with your new password."
+                      redirect={'/auth/login'}
+                      redirectText="Go to Login"
+                    />
+                  )
+                } else {
+                  const { errors, message } = response
+                  resetPasswordForm.setError('root', { message })
+
+                  if (errors) {
+                    getObjectKeys(errors).map((field) =>
+                      resetPasswordForm.setError(field, {
+                        message: errors[field]
+                      })
+                    )
+                  }
+                }
+              }}
+            >
+              <Input
+                required
+                type="password"
+                {...resetPasswordForm.register('password')}
+                error={resetPasswordForm.formState.errors.password?.message}
+                label="Password"
+              />
+              <Input
+                required
+                type="password"
+                {...resetPasswordForm.register('confirmPassword')}
+                error={
+                  resetPasswordForm.formState.errors.confirmPassword?.message
+                }
+                label="Confirm password"
+              />
+              <Input
+                required
+                type="hidden"
+                {...resetPasswordForm.register('token')}
+              />
+              <div className="flex justify-center py-5">
+                <Button
+                  aria-label="Reset Password"
+                  type="submit"
+                  loading={resetPasswordForm.formState.isSubmitting}
+                >
+                  Reset
+                </Button>
+              </div>
+            </Form>
+          </div>
+        </>
+      ) : (
+        <h2 className="mb-5 mt-10 text-center text-xl font-semibold text-green">
+          You are trying to access an invalid link. Please verify your link, or{' '}
+          <Link href="/auth/forgot" className="font-medium underline">
+            request link
+          </Link>{' '}
+          to reset password again.
+        </h2>
+      )}
+
+      <Image
+        className="mt-auto object-cover md:hidden"
+        src="/welcome-mobile.webp"
+        alt="Forgot password"
+        quality={100}
+        width={400}
+        height={200}
+      />
+      <p className="mt-auto font-extralight text-green">
+        Remembered your credentials?{' '}
+        <Link href="/auth/login" className="font-medium underline">
+          Login
+        </Link>
+      </p>
+    </>
+  )
+}
+
+const querySchema = z.object({
+  token: z.string()
+})
+
+export const getServerSideProps: GetServerSideProps<{
+  token: string
+}> = async (ctx) => {
+  const result = querySchema.safeParse(ctx.query)
```

Comment:
We should have a `check-token` controller to verify if the token exists or not. At the moment, any token will render the reset password page.